Add explicit text/plain template engine

gitblit opened the pull request:
This change allows you to return something like this:

```
Results.text().render(object);
```

The object's `toString()` will be invoked by the `TemplateEngineText` class.

This is more natural than the currently supported syntax:

```
Results.text().renderRaw(object.toString());
```

and fits in better with the other stock return types:

```
Results.json().render(object);
Results.jsonp().render(object);
Results.html().render(object);
Results.xml().render(object);
```

With the current state of _develop_ the former syntax will throw an exception because of the unbound template engine for `text/plain`.

gitblit committed:
Add explicit text/plain template engine

buildhive commented:
[Ninja Web Framework » ninja #180](https://buildhive.cloudbees.com/job/ninjaframework/job/ninja/180/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

raphaelbauer commented:
thanks :) will have a look the next days!

raphaelbauer commented:
That's pretty nice and clever :) Can you add the feature to changelog.md? I'll merge asap.

Btw - you can also deprecate the renderRaw(...) method if you want.

gitblit committed:
Add `text/plain` engine changelog entry

Also deprecate renderRaw(String) and add a Result.text() method to complement the other content type methods.

gitblit commented:
- added changelog entry
- updated JavaDoc
- deprecated Result.renderRaw(String)
- add Result.text() method to complement existing Result.contentTypeMethods

buildhive commented:
[Ninja Web Framework » ninja #185](https://buildhive.cloudbees.com/job/ninjaframework/job/ninja/185/) SUCCESS
This pull request looks good
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

raphaelbauer commented:
Awesome :+1:
